server/src/main/java/io/whitefox/core/types/predicates/LeafOp.java

```diff
@@ -0,0 +1,142 @@
+package io.whitefox.core.types.predicates;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.whitefox.core.types.BooleanType;
+import io.whitefox.core.types.DataType;
+import org.apache.commons.lang3.tuple.Pair;
+
+//import java.beans.ConstructorProperties;
+import java.util.List;
+import java.util.Objects;
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        property = "op"
+)
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = ColumnOp.class, name = "column"),
+        @JsonSubTypes.Type(value = LiteralOp.class, name = "literal")}
+)
+//@JsonDeserialize(using = LeafOpDeserializer.class)
+public abstract class LeafOp implements BaseOp {
+
+    abstract Boolean isNull(EvalContext ctx);
+
+    @JsonDeserialize(using = DataTypeDeserializer.class)
```

Comment:
Instead of setting this here in the field, can we configured it globally in `io.whitefox.api.configuration.JsonConfiguration#objectMapper` as follows?

```
var customSerializersModule = new SimpleModule();
customSerializersModule.addDeserializer(DataType.class, new DataTypeDeserializer());
mapper.registerModule(customSerializersModule);
```